### PR TITLE
Homework javascript-javascript3-week3

### PR DESCRIPTION
**Pull request by SvitlanaBurlaka**

### CODE REVIEW

File: javascript/javascript3/week3/Promises/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+function getData3() {
+    fetch("https://api.github.com/search/repositories?q=user:rebeccahockley")
+        .then((response) => response.json())
+        .then((data) => renderData(data));
+}
+
+// getData3();
+
+Promise.all([getData1(), getData2(), getData3()]);
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/Promises/index.js:40]: Superb!!!! 😃

File: javascript/javascript3/week3/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  draw() {
+    const canvas = document.getElementById("circle");
+    canvas.style.width = "100%";
+    canvas.style.height = "100%";
+    const ctx = canvas.getContext("2d");
+    ctx.beginPath();
+    ctx.arc(this.x, this.y, this.r, this.startAngle, this.endAngle);
+    ctx.fillStyle = this.fillColor;
+    ctx.stroke();
+    ctx.fill();
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/index.js]: To make the class reusable, we can pass "context" as an argument to the draw method.
 
 const canvas = document.getElementById("circle");
    canvas.style.width = "100%";
    canvas.style.height = "100%";
    const ctx = canvas.getContext("2d");

draw(context) {
    ctx.beginPath();
    ctx.arc(this.x, this.y, this.r, this.startAngle, this.endAngle);
    ctx.fillStyle = this.fillColor;
    ctx.stroke();
    ctx.fill();
}

const c1 = new Circle(50, 50, 20, 0, 2 * Math.PI, "yellow");
c1.draw(ctx);

File: javascript/javascript3/week3/index.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+// Now lets make art!
+setInterval(() => {
+  const xRandom = Math.floor(Math.random() * 500);
+  const yRandom = Math.floor(Math.random() * 500);
+  const rRandom = Math.floor(Math.random() * 200);
+  var randomColor = "#" + Math.floor(Math.random() * 16777215).toString(16);
+
+  const c1 = new Circle(xRandom, yRandom, rRandom, 0, 2 * Math.PI, randomColor);
+  c1.draw();
+}, 100);
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/index.js]: 👍

File: javascript/javascript3/week3/shopping cart/app.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        this.products.splice(this.products.indexOf(product), 1);
+    }
+
+    searchProduct(productName) {
+        let searchedProduct = {};
+        this.products.forEach((product) => {
+            if (product.name === productName) {
+                searchedProduct = product;
+            }
+        });
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/shopping cart/app.js:45]: Great!!!
You can also use "filter" to search an item into an array.

File: javascript/javascript3/week3/shopping cart/app.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        return searchedProduct;
+    }
+
+    getTotal() {
+        let sum = 0;
+        for (let i = 0; i < this.products.length; i++) {
+            sum += this.products[i].price;
+        }
+        return sum;
+    }
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/shopping cart/app.js:55]: Using "reduce" is another good way.

 const sum= this.products.reduce((prev, next) => prev + next.price, 0);

File: javascript/javascript3/week3/index.js (hunk 2 above)
Comment (SvitlanaBurlaka) [reply javascript/javascript3/week3/index.js]: @ElhamDargi Hi. Thank you for review. Please, take a look , I changed this part.

File: javascript/javascript3/week3/index.js [hunk 6]
```diff
@@ -2,37 +2,38 @@
...
+        this.y = y;
+        this.r = r;
+        this.startAngle = startAngle;
+        this.endAngle = endAngle;
+        this.fillColor = fillColor;
+    }
+
+    draw(ctx) {
+        canvas.style.width = "100%";
+        canvas.style.height = "100%";
```
Comment (ElhamDargi) [inline javascript/javascript3/week3/index.js:16]: Excellent!!
It is better to set  "width & height"  where you get the "canvas" with "getElementById".

const canvas = document.getElementById("circle");
canvas.style.width = "100%";
canvas.style.height = "100%";

**Pull request merged**